### PR TITLE
NCSDK-26629: Migrate SUIT platform

### PR DESCRIPTION
**Pull request by tomchy**

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -343,6 +344,7 @@ Kconfig*                                  @tejlmand
 /tests/subsys/nrf_profiler/               @pdunaj @MarekPieta
 /tests/subsys/sdfw_services/              @anhmolt @hakonfam @jonathannilsen
 /tests/subsys/zigbee/                     @milewr
+/tests/subsys/suit/                       @tomchy
```
Comment (ahasztag) [inline CODEOWNERS:347]: Wouldn't we like to extend the code owners list with other developers in the SUIT team (probably @adsz-nordic, @ahasztag perhaps @e-rk if he'd like). I think it wouldn't block anyone, and the team would be informed about external changes if you'd be on vacation

File: subsys/mgmt/suitfu/src/suitfu_mgmt_img.c [hunk 2]
```diff
@@ -200,12 +200,13 @@ static int suitfu_mgmt_img_state_read(struct smp_streamer *ctx)
...
 	if (ok) {
 		installed_vers_str[sizeof(installed_vers_str) - 1] = '\0';
-		ok = zcbor_tstr_put_term(zse, installed_vers_str);
+		ok = zcbor_tstr_put_term(zse, installed_vers_str, sizeof(installed_vers_str));
 		LOG_DBG("Manifest version encoded: %d", ok);
 	}
 
 	if (ok) {
-		ok = zcbor_tstr_put_term(zse, "hash") && zcbor_bstr_encode(zse, &zhash);
+		ok = zcbor_tstr_put_term(zse, "hash", sizeof("hash")) &&
```
Comment (oyvindronningstad) [inline subsys/mgmt/suitfu/src/suitfu_mgmt_img.c:208]: All these literals can be encoded using `*_put_lit()` instead, to avoid the runtime call to strnlen.

**Review by adsz-nordic — APPROVED** [review]

**Review by robertstypa — APPROVED** [review]

**Review by oyvindronningstad — APPROVED** [review]

File: tests/subsys/suit/flash_sink/boards/nrf54h20dk_nrf54h20_cpuapp.overlay [hunk 3]
```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+&mram1x {
+	erase-block-size = < 0x400 >;
```
Comment (nordicjm) [inline tests/subsys/suit/flash_sink/boards/nrf54h20dk_nrf54h20_cpuapp.overlay:8]: Not going to request changes but might be an idea to have one of these files and then just `#include "../..."` from the rest

File: subsys/suit/memory_layout/Kconfig [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 config SUIT_MEMORY_LAYOUT_EXTMEM_ADDRESS_RANGE_START
 	hex
-	default 0x60000000 if SOC_NRF54H20
+	default 0x60000000 if SOC_SERIES_NRF54HX
 	default 0x12000000 if SOC_NRF52840
 	default 0x0 if BOARD_NATIVE_POSIX
```
Comment (nordicjm) [inline subsys/suit/memory_layout/Kconfig:11]: This should come from dts

File: subsys/suit/memory_layout/Kconfig [hunk 5]
```diff
@@ -15,7 +15,7 @@ config SUIT_MEMORY_LAYOUT_EXTMEM_ADDRESS_RANGE_START
 
 config SUIT_MEMORY_LAYOUT_EXTMEM_ADDRESS_RANGE_SIZE
 	hex
-	default 0x40000000 if SOC_NRF54H20
+	default 0x40000000 if SOC_SERIES_NRF54HX
 	default 0x8000000 if SOC_NRF52840
 	default 0x0 if BOARD_NATIVE_POSIX
 	help
```
Comment (nordicjm) [inline subsys/suit/memory_layout/Kconfig:21]: and this, this is a hardware property which should come from dts

File: subsys/suit/stream/Kconfig [hunk 6]
```diff
@@ -44,14 +44,14 @@ if SUIT_STREAM_SINK_EXTMEM
...
 	  Start address of the extended memory range.
 	  This value is SOC specific and is not meant to be changed.
 
 config SUIT_STREAM_SINK_EXTMEM_ADDRESS_RANGE_SIZE
 	hex
-	default 0x40000000 if SOC_NRF54H20
+	default 0x40000000 if SOC_SERIES_NRF54HX
 	help
 	  Size of the extended memory range.
 	  This value is SOC specific and is not meant to be changed.
```
Comment (nordicjm) [inline subsys/suit/stream/Kconfig:57]: I don't get why this is here, you already have a Kconfig (which needs to move to dts) but why is the option duplicated already?

Comment (tomchy) [reply subsys/suit/stream/Kconfig:57]: Nice catch!
This is a leftover and can be safely removed.

File: subsys/suit/memory_layout/Kconfig (hunk 4 above)
Comment (tomchy) [reply subsys/suit/memory_layout/Kconfig:11]: Unfortunately the EXMIF is not yet defined in DTS. To be aligned in the future (possibly along with the falsh_companion sample migration process).

File: subsys/suit/memory_layout/Kconfig (hunk 5 above)
Comment (tomchy) [reply subsys/suit/memory_layout/Kconfig:21]: Same here - I'll add a note in the task describing flash_companion image migration.

**Review by nordicjm — APPROVED** [review]

**Pull request merged**